frontend/ui/src/app/deployments/deployment-status-modal/timeseries-table.component.ts

```diff
@@ -132,6 +156,55 @@ export class TimeseriesTableComponent {
   protected showMore() {
     this.showMore$.next();
   }
+
+  protected exportData() {
+    const deploymentId = this.deploymentId();
+    const exportType = this.exportType();
+
+    if (!deploymentId || !exportType) {
+      return;
+    }
+
+    this.isExporting = true;
+
+    let exportObservable: Observable<Blob>;
+    let filename: string;
+    const today = new Date().toISOString().split('T')[0];
+
+    if (exportType === 'logs') {
+      const resource = this.resource();
+      if (!resource) {
+        this.isExporting = false;
+        return;
+      }
+      exportObservable = this.deploymentLogsService.export(deploymentId, resource);
+      filename = `${today}_${resource}.log`;
+    } else if (exportType === 'status') {
+      exportObservable = this.deploymentStatusService.export(deploymentId);
+      filename = `${today}_deployment_status.log`;
+    } else {
+      this.isExporting = false;
+      return;
+    }
+
+    exportObservable.subscribe({
+      next: (blob) => {
+        const url = window.URL.createObjectURL(blob);
```

Comment:
It would be really cool if we could also "stream" the download on the client side by using the browsers builtin download manager instead of buffering the blob. I assume this is done because auth would be missing otherwise, but OTOH it would be very easy to add support for authenticating via query param in the backend. A token extractor func already exists for this.